aip/general/0211.md

```diff
@@ -0,0 +1,55 @@
+---
+id: 211
+state: reviewing
+created: 2021-02-24
+placement:
+  category: design-patterns
+  order: 115
+---
+
+# Authorization checks
+
+The majority of operations, whether reads or writes, require authorization:
+permission to do the thing the user is asking to do. Additionally, it is
+important to be careful how much information is provided to _unauthorized_
+users, since leaking information can be a security concern.
+
+## Guidance
+
+Services **must** check authorization before validating any request, to ensure
+both a secure API surface and a consistent user experience. An operation
+**may** require multiple permissions or preconditions in order to grant
+authorization.
+
+If a request can not pass the authorization check for any reason, the service
+**must** error with `PERMISSION_DENIED`, and the corresponding error message
+**should** look like: "Permission `{p}` denied on resource `{r}` (or it might
+not exist)." This avoids leaking resource existance.
+
+**Note:** This guidance conflicts with the guidance in the HTTP specification,
+which argues for the use of 404 in cases where the user lacks permission to
+know whether or not a resource exists. This variance is because of real-world
+challenges: the practice of "getting not found until you have enough permission
+to get permission denied" is counter-intuitive and increases the difficulty of
+troubleshooting. Additionally, the guidance here is more consistent with most
+other authorization systems.
+
+If it is not possible to determine authorization for a resource because the
+resource does not exist, the service **should** check authorization on the
+parent resource (or, if the system permits such a concept, for a `null`
+resource under the same parent), and return `NOT_FOUND` if the authorization
+check passes.
+
+### Multiple operations
```

Comment:
I think there is important information here but I'm not following what the recommendation is. Can we maybe use a specific example to clarify?

Reply:
Your example above is actually the example. Will add.

Reply:
Done.